src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -522,8 +522,17 @@ size_t TieredHNSWIndex<DataType, DistType>::indexLabelCount() const {
 template <typename DataType, typename DistType>
 int TieredHNSWIndex<DataType, DistType>::addVector(const void *blob, labelType label,
                                                    void *auxiliaryCtx) {
-    /* Note: this currently doesn't support overriding (assuming that the label doesn't exist)! */
+    int ret = 1;
     this->flatIndexGuard.lock();
+    if (this->frontendIndex->isLabelExists(label) && !this->backendIndex->isMultiValue()) {
+        // Overwrite the vector and invalidate its only pending job (since we are not in MULTI).
+        this->frontendIndex->deleteVector(label);
+        auto *old_job = this->labelToInsertJobs.at(label).at(0);
+        old_job->id = INVALID_JOB_ID;
+        this->labelToInsertJobs.erase(label);
+        ret = 0;
+    }
+
     if (this->frontendIndex->indexCapacity() == this->frontendIndex->indexSize()) {
```

Comment:
I guess this can be `else if` now, as we either need to make sure we have the capacity or we just deleted a vector so there is surely capacity for the new vector